docs/specification/draft/basic/transports.md

```diff
@@ -48,38 +48,209 @@ sequenceDiagram
+     - The SSE stream **SHOULD** eventually include a JSON-RPC _response_ message.
+     - The server **MAY** send JSON-RPC _requests_ and _notifications_ before sending a
+       JSON-RPC _response_. These messages **SHOULD** relate to the originating client
+       _request_.
+     - The server **SHOULD NOT** close the SSE stream before sending the JSON-RPC
+       _response_, unless the [session](#session-management) expires.
+     - After the JSON-RPC _response_ has been sent, the server **MAY** close the SSE
+       stream at any time.
+     - Disconnection **MAY** occur at any time (e.g., due to network conditions).
+       Therefore:
+       - Disconnection **SHOULD NOT** be interpreted as the client cancelling its
+         request.
+       - To cancel, the client **SHOULD** explicitly send an MCP `CancelledNotification`.
+       - To avoid message loss due to disconnection, the server **MAY** make the stream
+         [resumable](#resumability-and-redelivery).
+
+3. When the client sends a JSON-RPC _notification_ or _response_ to the MCP endpoint via
+   POST:
+
+   - If the server accepts the message, it **MUST** return HTTP status code 202 Accepted
+     with no body.
+   - If the server cannot accept the message, it **MUST** return an HTTP error status
+     code (e.g., 400 Bad Request). The HTTP response body **MAY** comprise a JSON-RPC
+     _error response_ that has no `id`.
+
+4. The client **MAY** also issue an HTTP GET to the MCP endpoint. This can be used to
+   open an SSE stream, allowing the server to communicate to the client without the
+   client first sending a JSON-RPC _request_.
+   - The client **MUST** include an `Accept` header, listing `text/event-stream` as a
+     supported content type.
+   - The server **MUST** either return `Content-Type: text/event-stream` in response to
+     this HTTP GET, or else return HTTP 405 Method Not Allowed, indicating that the
+     server does not offer an SSE stream at this endpoint.
+   - If the server initiates an SSE stream:
+     - The server **MAY** send JSON-RPC _requests_ and _notifications_ on the stream.
+       These messages **SHOULD** be unrelated to any concurrently-running JSON-RPC
+       _request_ from the client.
+     - The server **MUST NOT** send a JSON-RPC _response_ on the stream **unless**
+       [resuming](#resumability-and-redelivery) a stream associated with a previous
+       client request.
+     - The server **MAY** close the SSE stream at any time.
+     - The client **MAY** close the SSE stream at any time.
+
+### Multiple Connections
+
+1. The client **MAY** remain connected to multiple SSE streams simultaneously.
+2. The server **MUST** send each of its JSON-RPC messages on only one of the connected
+   streams; that is, it **MUST NOT** broadcast the same message across multiple streams.
+   - The risk of message loss **MAY** be mitigated by making the stream
+     [resumable](#resumability-and-redelivery).
+
+### Resumability and Redelivery
```

Comment:
As long as you support resumability and redelivery, most stateful/persistent-connection issues associated with scaling will disappear, with or without the rest of the proposed changes in this PR. 

This alone will allow the server to safely drop the SSE connection whenever necessary. The client can then retrieve notifications by engaging in a long-polling or shorter-lived regular interval polling mechanism.

This will also help intentionally stateful client/server integrations remain stable during server software upgrades and scaling/rebalancing events.

This is such an essential stand-alone feature/change that it should be broken out as a much smaller change and integrated into the existing SSE-based transport as it is.

Reply:
There is nothing prohibiting SSE resumption in the current SSE transport. Given how minimally it's specified for MCP, and the fact that resumability is a part of SSE's spec, I don't see any need to add additional mentions to the MCP spec for the (now deprecated) SSE transport.